src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import java.util.Scanner;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.layout.Region;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+public class Duke {
+    private static String taskDataPath = "data";
+    private static String taskDataFileName = "duke.txt";
+
+    private static Scanner sc = new Scanner(System.in);
+    private static UI UI = new UI();
+    private static Parser parser = new Parser();
+    private static TaskList taskList = new TaskList();
+    private static Storage storage = new Storage(taskDataPath, taskDataFileName);
+
+    Duke() {
+        loadStorage();
+    }
+
+
+    private static String loadStorage() {
+        try {
+            storage.readTaskData(taskList);
+            return "";
+        } catch (DukeException e) {
+            return e.getMessage();
+        }
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
+        String response = "";
+
+        switch (input) {
+            case "list":
+            case "list\n":
+                response = taskList.printTaskList();
+                break;
+            case "bye":
+            case "bye\n":
+                response = UI.goodBye;
+                storage.updateTaskData(taskList);
+                break;
+            default:
+                try {
+                    String[] commands = input.split(" ");
+
+                    if (commands.length < 2) {
```

Comment:
You might want to consider using a nested switch statements instead of so many if-else statements.